api/v1beta1/azuremanagedcontrolplane_webhook.go

```diff
@@ -669,3 +676,38 @@ func (m *AzureManagedControlPlane) validateIntegerStringGreaterThanZero(input *s
 
 	return allErrs
 }
+
+// validateIdentityProfile validates an AutoScalerProfile.
+func (m *AzureManagedControlPlane) validateIdentityProfile(_ client.Client) error {
+	var allErrs field.ErrorList
+
+	if m.Spec.IdentityProfile != nil {
+		if m.Spec.IdentityProfile.Type == ManagedControlPlaneIdentityTypeUserAssigned {
+			if m.Spec.IdentityProfile.UserAssignedIdentityResourceID == "" {
+				allErrs = append(allErrs, field.Invalid(field.NewPath("Spec", "IdentityProfile", "UserAssignedIdentityResourceID"), m.Spec.IdentityProfile.UserAssignedIdentityResourceID, "cannot be empty if IdentityProfile.Type is UserAssigned"))
+			}
+			if m.Spec.IdentityProfile.UserAssignedIdentityPrincipalID == nil {
+				allErrs = append(allErrs, field.Invalid(field.NewPath("Spec", "IdentityProfile", "UserAssignedIdentityPrincipalID"), m.Spec.IdentityProfile.UserAssignedIdentityPrincipalID, "cannot be empty if IdentityProfile.Type is UserAssigned"))
+			}
+			if m.Spec.IdentityProfile.UserAssignedIdentityClientID == nil {
+				allErrs = append(allErrs, field.Invalid(field.NewPath("Spec", "IdentityProfile", "UserAssignedIdentityClientID"), m.Spec.IdentityProfile.UserAssignedIdentityClientID, "cannot be empty if IdentityProfile.Type is UserAssigned"))
+			}
+		} else {
+			if m.Spec.IdentityProfile.UserAssignedIdentityResourceID != "" {
+				allErrs = append(allErrs, field.Invalid(field.NewPath("Spec", "IdentityProfile", "UserAssignedIdentityResourceID"), m.Spec.IdentityProfile.UserAssignedIdentityResourceID, "should be empty if IdentityProfile.Type is SystemAssigned"))
+			}
+			if m.Spec.IdentityProfile.UserAssignedIdentityPrincipalID != nil {
```

Comment:
I don't even think you should expose these fields except in status. but see above. 

if I'm mistaken, this makes sense to me.